tests/integration/engines/test_int_functions.py

```diff
@@ -2911,7 +3036,10 @@ def test_transform_keys(get_session_and_func, get_func):
     upper = get_func("upper", session)
     df = session.createDataFrame([(1, {"foo": -2.0, "bar": 2.0})], ("id", "data"))
     row = df.select(transform_keys("data", lambda k, _: upper(k)).alias("data_upper")).head()
-    assert sorted(row["data_upper"].items()) == [("BAR", 2.0), ("FOO", -2.0)]
+    if isinstance(session, DatabricksSession):
+        assert sorted(row["data_upper"]) == [["BAR", 2.0], ["FOO", -2.0]]
```

Comment:
Can you provide some context as to why this should be diffferent? It seems like the key difference is that `data_upper` isn't a dict in Databricks. Is that expected? Is there a way to have is conform to the expected data type?

Reply:
Hi,
I've found that the databricks sql connector has a couple of "problems/differences" compared with the rest of connectors.

It has different type conversions for some data types.

These are the differences:
The MAP type gets converted to a list of lists instead of a dictionary.
The STRUCT type gets converted to a dictionary instead of a ROW.

I haven't found anything in the connector to change this behaviour but I'll try to look at it a little deeper.

Reply:
Hi,

I've kind of solved this but there's a new problem. the `_to_value` method transform all the dict keys to lowercase

Reply:
See here: https://github.com/eakmanrq/sqlframe/pull/218#issuecomment-2540570341

In short this is some nuance related to normalization and it is fine for now.

Reply:
Hi, I've changed the test for databricks to append a character instead og the upper function.